2025 H2 Roadmap (#311)
waku-org/pm: draft-roadmap/README.md

# Waku Draft Roadmap

**Most of the below is being moved to [2025H2 Summary](../2025H2-summary.md)**, which is now the master document.

Finalised roadmap and milestones can be found on the [Logos roadmap](https://roadmap.logos.co/waku/).

Period in planning: 2025 H2

## Proposed Priorities

### Whole Team

1. **Support Status application**: continue integration of nwaku and provide new chat protocol stack. 
2. **Support Logos Core**: ensure that any library and API we push to developers can be consumed in Logos Core; review and prioritize any requirements coming from Logos Core project contributors and developers.
3. **Developer and Contributor Experience:** Review and improve dev assets (docs) and set quality expectation across the team (docs provided with deliverables, etc); Review and take action to improve developer and contributor experience (nimble usage, vscode plugin, etc).

### Core Team

1. **Complete integration of nwaku in Status Desktop** (H1)

> 2. **Simplify a reliable Waku API:** aka Messaging API, make it easier to consume Waku library; critical to enable easy development of Chat SDK.

Reviewer: ```suggestion
2. **Simplify a reliable Waku SDK:** aka Messaging API, make it easier to consume Waku library; critical to enable easy development of Chat SDK.
```

   1. Potentially expand the relevant platforms: from Web to Logos core, Nim/seaqt, Rust, etc.
   2. Integration with Codex still planned under this umbrella
   3. Review Logos Forum (Opchan) requirements; the 2025 Logos Movement Strategy is the new customer.

### Business and Ecosystem Development Team (BD + Solution Eng)

1. **Measure Success:** Develop and deploy tools to measure Waku success in terms of users, developers and contributors across all known Waku networks (Status, RAILGUN, TWN, etc)
    1. In partnership with BI.
2. **Waku Chat MVP:** Proceed with the same exercise done for [Waku MVP](https://www.notion.so/Waku-MVP-1838f96fb65c8039acabf8a6a1e689e7?pvs=21).
   Consult with current and new leads on their *ideal* Chat SDK.
   Understand how confident we are they would onboard on Waku if it is delivered and feedback to Chat/App team to take in account for prioritization.
3. **Support integrations**: support projects that want to build with Waku SDK and Chat SDK. 
4. **Foster and join Nim ecosystem (nwaku):** Foster and participate in the Nim developer community inside and outside IFT.
5. **Join FOSS ecosystem:** Be an active part of the FOSS developer community.

> 6. **Continue planned Rust SDK work**: Messaging API and stable nwaku integration in Status Desktop and pre-requisites to a quality Rust SDK.

Reviewer: ```suggestion
6. **Continue planned Rust SDK work**: Waku SDK and stable nwaku integration in Status Desktop and pre-requisites to a quality Rust SDK.
```

| [Foundation for Communities Optimisation](foundation_for_communities_optimisation.md)                      | 31 Oct         |          |         |       | 1        | 
| [Hardening and Scaling Foundations for Private Chat](hardening_and_scaling_foundation_for_private_chat.md) | Completed/Drop |          |         |       |          | 
| [Integrate nwaku in Status Desktop, relay mode only](integrate_nwaku_in_status_desktop_relay_mode_only.md) | 30 Jun         |          |         | 2     |          |
| [Deploy RLN Onchain Tree on L2 Testnet](deploy_rln_onchain_tree_on_l2_testnet.md)                          | 30 Jun         |          |         |       |          |

### H2 Milestones

| Priority | Milestone                                                                               | End Date | core res | js-waku | nwaku  | app-chat |
|----------|-----------------------------------------------------------------------------------------|----------|----------|---------|--------|----------|
| 1        | [Define Incentivisation for RLNaaS](define_incentivisation_for_rlnaas.md)               | 31 Jul   | 1.5*1m   |         |        |          |

> | 2        | [Improve DevEx: API, TWN, Metrics, Docs](improve_devex_api_twn_metrics_docs.md)         | 31 Aug   | 1*1m     | 2*2m    | 1.5*2m | 1*1m     |

Reviewer: ` 1*1m     | 2*2m    | 1.5*2m | 1*1m `

The way this is expresses is really awesome! Maybe interesting having a little legend somewhere for external readers

| 5        | [Create Chat SDK MVP](create_chat_sdk_mvp.md)                                           | 30 Sep   |          |         |        | 3*3m     |
| 6        | [Integrate RLN with Waku API](integrate_rln_with_waku_api.md)                           | 30 Sep   | 1*1m     | 1*2m    | 1*2m   |          |
| 7        | [Streamline DevEx: Mobile, Rust and Web dev](streamline_dev_ex_local_dev_rust.md)       | 30 Nov   |          | 1*1m    | 1*2m   |          |
| 8        | [Extend Chat SDK with Group Conversations](extend_chat_sdk_with_group_conversations.md) | 19 Dec   |          |         |        | 2*3m     |
| 9        | [Incentivisation follow-up Outline](incentivisation_follow_up.md)                       | TBD      | 1.5*5m   |         |        |          |
| 10       | [Nim Usage Improvements](nim_usage_improvements.md)                                     | 19 Dec   |          |         | 1*2m   |          |
| 11       | [BD - Acquire 10 Customers](acquire_first_10_customers.md)                              | 19 Dec   |          |         |        | 0.5*6m   | 

Pushed to 2026
- WebTransport: depending on nim-libp2p (delivery Q4)

> - REST API for Waku API: Useful for DST/QA, but let's focus on Status, Chat SDK, and Rust first

Reviewer: ```suggestion
- REST API for Waku SDK: Useful for DST/QA, but let's focus on Status, Chat SDK, and Rust first
```

Not yet planned/not sure:
- nwaku performance on mobile: let's focus on finishing desktop integration and getting API ready for Chat SDK + RLN
- Follow-up steps for incentivization: part of current milestone is to produce a roadmap.
- RLN mainnet and audit -> probably wait for Status L2 mainnet?
- Signal network

## Gantt

TODO: fix dates

```mermaid
gantt
    title Waku 2025H2
    dateFormat YYYY-MM-DD
    axisFormat %b
    section core research (6)
        Define Incentivization for RLNaaS: 2025-07-01, 2025-08-01
        Define Incentivization for RLNaaS: 2025-07-01, 2025-08-01
        Improve DevEx: 2025-08-01, 2025-09-30
        Mixnet: 2025-07-01, 2025-09-30
        Integrate RLN with Waku API: 2025-07-01, 2025-09-01
        Streamline DevEx: 2025-09-01, 2025-12-31
        Incentivization follow-up: 2025-10-01, 2025-12-31
    section nwaku
        Status Desktop: 2025-07-01, 2025-08-01
        RLN Onchain Tree: 2025-07-01, 2025-08-01

>         Improve DevEx (API): 2025-07-01, 2025-10-01

Reviewer: Maybe?
```suggestion
        Improve DevEx (SDK): 2025-07-01, 2025-10-01
```
